### PR TITLE
Searchfeature clean

### PR DESCRIPTION
**Pull request by jgarza9788**
copy, paste, test ... almost forgot clang-format.sh
and now submitting again

### CODE REVIEW

**Comment by jgarza9788** [comment]
>.< forgot to save a file

File: resources/ui/adw/prefs.ui [hunk 1]
```diff
@@ -214,39 +214,132 @@ SPDX-License-Identifier: GPL-3.0-or-later
     </child>
   </object>
 
+
   <object class="AdwPreferencesPage" id="general-prefs">
+    <property name="margin-top">20</property>
+    <property name="margin-bottom">20</property>
+    <property name="margin-start">20</property>
+    <property name="margin-end">20</property>
```
Comment (Schneegans) [inline resources/ui/adw/prefs.ui:222]: I think these are not required.

```suggestion
```

File: resources/ui/adw/prefs.ui [hunk 2]
```diff
@@ -214,39 +214,132 @@ SPDX-License-Identifier: GPL-3.0-or-later
...
           </object>
         </child>
+
+      </object>
+    </child>
+
+    <child>
+      <object class="AdwPreferencesGroup" id="effects-group">
+        <property name="margin-top">10</property>
+        <property name="margin-bottom">0</property>
```
Comment (Schneegans) [inline resources/ui/adw/prefs.ui:335]: I think this needs some padding at the bottom now.

```suggestion
        <property name="margin-bottom">20</property>
```

File: resources/ui/adw/prefs.ui [hunk 3]
```diff
@@ -214,39 +214,132 @@ SPDX-License-Identifier: GPL-3.0-or-later
...
-        <property name="description" translatable="yes">If multiple are selected, one is chosen randomly.</property>
-
-        <child type="header-suffix">
-          <object class="GtkBox">
-            <property name="valign">center</property>
-            <style>
-              <class name="linked" />
-            </style>
+      <object class="AdwPreferencesGroup" id="">
+
```
Comment (Schneegans) [inline resources/ui/adw/prefs.ui:226]: Some things are kind of weird here. I think having a preferences group inside a preferences group is not desirable. I think you can simply move the title and description to the outer group and remove the inner group.

Add this:

```suggestion
      <object class="AdwPreferencesGroup">
         <property name="title" translatable="yes">Effect Selection</property>
         <property name="description" translatable="yes">If multiple are selected, one is chosen randomly.</property>
         
```

And completely remove the preferences group `header-text`.

File: resources/ui/adw/prefs.ui [hunk 4]
```diff
@@ -214,39 +214,132 @@ SPDX-License-Identifier: GPL-3.0-or-later
...
 
-            <child>
-              <object class="GtkButton" id="choose-no-effects-button">
-                <property name="label" translatable="yes">None</property>
-              </object>
-            </child>
+          <!-- Search Entry -->
+          <child>
+            <object class="GtkSearchEntry" id="search_entry">
+              <property name="placeholder-text">Search...</property>
```
Comment (Schneegans) [inline resources/ui/adw/prefs.ui:247]:
```suggestion
              <property name="placeholder-text" translatable="yes">Search...</property>
```

**Review by Schneegans — CHANGES_REQUESTED** [review]
Thanks a lot! It works well but I think the `prefs.ui` file will need some clean up.

Btw, a handy one-liner to restart the settings dialog to preview any changes is

```bash
make install && pkill -f '.Extensions' && sleep 0.5 ; gnome-extensions prefs burn-my-windows@schneegans.github.com
```

File: resources/ui/adw/prefs.ui (hunk 1 above)
Comment (jgarza9788) [reply resources/ui/adw/prefs.ui:222]: ummm i think these were in the main code, so i wasn't exactly sure if it was needed.
👍

**Comment by jgarza9788** [comment]
one of the changes from above caused there to be two pieces of text,
lets see if i can solve this
image

**Pull request closed without merging**